sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/SaveAsHiveFile.scala

```diff
@@ -124,11 +162,23 @@ private[hive] trait SaveAsHiveFile extends DataWritingCommand {
     val hiveVersion = externalCatalog.unwrapped.asInstanceOf[HiveExternalCatalog].client.version
     val stagingDir = hadoopConf.get("hive.exec.stagingdir", ".hive-staging")
     val scratchDir = hadoopConf.get("hive.exec.scratchdir", "/tmp/hive")
+    logDebug(s"path '${path.toString}', staging dir '$stagingDir', " +
+      s"scratch dir '$scratchDir' are used")
 
     if (hiveVersionsUsingOldExternalTempPath.contains(hiveVersion)) {
       oldVersionExternalTempPath(path, hadoopConf, scratchDir)
     } else if (hiveVersionsUsingNewExternalTempPath.contains(hiveVersion)) {
-      newVersionExternalTempPath(path, hadoopConf, stagingDir)
+      // HIVE-14270: Write temporary data to HDFS when doing inserts on tables located on S3
+      // Copied from Context.java#getTempDirForPath of Hive 2.3
+      if (isBlobStoragePath(path) && !useBlobStorageAsScratchDir) {
+        // Hive sets session_path as HDFS_SESSION_PATH_KEY(_hive.hdfs.session.path) in hive config
+        val sessionScratchDir = externalCatalog.unwrapped.asInstanceOf[HiveExternalCatalog]
+          .client.getConf("_hive.hdfs.session.path", "")
+        logDebug(s"session scratch dir '$sessionScratchDir' is used")
```

Comment:
Looks this logging is similar to those inside `getMRTmpPath`.
https://github.com/apache/spark/pull/27690/files#diff-e08e0d262de1e7582ddee984936d93d4R118
https://github.com/apache/spark/pull/27690/files#diff-e08e0d262de1e7582ddee984936d93d4R121
Could you merge them in a single place? I think It is hard to read the current logs.

Reply:
Deleted the debug logging.